woocommerce/woocommerce-gutenberg-products-block
# Remove vendors.js as a frontend dependency

Depends on #746.

This PR:
* Removes `vendors.js` as a frontend dependency.
* Uses externals for dependencies, so they are shared between plugins and blocks.
* Merges `packages.js` and `vendors.js` files. They were split so `vendors.js` could be included in the frontend, but that is no longer necessary.

### How to test the changes in this Pull Request:

Test `vendors.js` is no longer a dependency:
1. Create a post and add a _Product Categories List_ block.
2. Open the post in the frontend and verify it works as expected.
3. In the _Network_ tab of your browser devtools, verify no `vendors.js` file was loaded.

Test we are using externals
1. Install Jetpack.
2. Create a post and add a _Product Categories List_ block, a _Slideshow_ block and a _Map_ block.
3. Open the post in the frontend and, in the _Network_ tab of your browser devtools, verify only one `lodash.js` file was loaded.

### Changelog

Use externals for frontend dependencies so they are shared between extensions and blocks. That saves 2.57MB on page weight.
